modules/apps/frontend-data-set/frontend-data-set-impl/src/main/java/com/liferay/frontend/data/set/internal/url/CustomFDSAPIURLSerializerImpl.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set.internal.url;
+
+import com.liferay.frontend.data.set.constants.FDSTypes;
+import com.liferay.frontend.data.set.internal.serializer.CustomFDSSerializerHelper;
+import com.liferay.frontend.data.set.url.FDSAPIURLBuilder;
+import com.liferay.frontend.data.set.url.FDSAPIURLBuilderFactory;
+import com.liferay.frontend.data.set.url.FDSAPIURLSerializer;
+import com.liferay.object.rest.dto.v1_0.ObjectEntry;
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Daniel Sanz
+ */
+@Component(
+	property = "dataset.type=" + FDSTypes.CUSTOM,
+	service = FDSAPIURLSerializer.class
+)
+public class CustomFDSAPIURLSerializerImpl implements FDSAPIURLSerializer {
+
+	@Override
+	public String serialize(
+		String fdsName, HttpServletRequest httpServletRequest) {
+
+		Map<String, Object> properties =
+			_customFDSSerializerHelper.getDataSetObjectEntryProperties(
+				fdsName, httpServletRequest);
+
+		FDSAPIURLBuilder fdsAPIURLBuilder = _fdsAPIURLBuilderFactory.create(
+			httpServletRequest,
+			String.valueOf(properties.get("restApplication")),
+			String.valueOf(properties.get("restEndpoint")),
+			String.valueOf(properties.get("restSchema")));
+
+		return _addNestedFields(
+			fdsAPIURLBuilder,
+			_customFDSSerializerHelper.getDataSetTableSectionObjectEntries(
+				fdsName, httpServletRequest)
+		).build();
+	}
+
+	private FDSAPIURLBuilder _addNestedFields(
+		FDSAPIURLBuilder fdsapiurlBuilder,
+		Set<ObjectEntry> dataSetTableSectionObjectEntries) {
+
+		if (dataSetTableSectionObjectEntries == null) {
+			return fdsapiurlBuilder;
+		}
+
+		String nestedFields = StringPool.BLANK;
+		int nestedFieldsDepth = 1;
+
+		for (ObjectEntry fdsFieldObjectEntry :
+				dataSetTableSectionObjectEntries) {
+
+			Map<String, Object> properties =
+				fdsFieldObjectEntry.getProperties();
+
+			String[] fieldNameList = StringUtil.split(
+				StringUtil.replace(
+					String.valueOf(properties.get("fieldName")), "[]",
+					StringPool.PERIOD),
+				CharPool.PERIOD);
+
+			if (fieldNameList.length > 1) {
+				String[] fieldsName = new String[fieldNameList.length - 1];
+
+				System.arraycopy(
+					fieldNameList, 0, fieldsName, 0, fieldNameList.length - 1);
```

Comment:
Without going too deep into what's going on here, there is probably a nicer way to achieve it, a util or a tweak when making an array instance.